src/main/webapp/teacher/controlGameInstance/controlGameInstance.js

```diff
@@ -442,4 +444,96 @@ function addQuestionAnswerToHistoryUI({ questionId, answerId, answer } = {}) {
+        initStudentHistory({ studentId: change.doc.id, student: change.doc.data(), studentsCollectionRef });
+      }
+
+      // If it was modified, a student answered a question
+      if (change.type === "modified") {
+        // Just update the stats of student
+        updateStudentStats({ studentId: change.doc.id, student: change.doc.data() });
+      }
+    });
+  });
+}
+
+// Add student data to UI and startt listening to his "questions" collection
+function initStudentHistory({ studentId, student, studentsCollectionRef } = {}) {
+  addStudentToUI({ studentId, student });
+  initStudentQuestionListener({ studentId, studentsCollectionRef });
+}
+
+// Add student general stats to UI
+function addStudentToUI({ studentId, student } = {}) {
+  const studentsStatsDivElement = document.getElementById('jsStudentStats');
+
+  const newStudentStatsDivElement = document.createElement('div');
+  newStudentStatsDivElement.id = 'studentStats-' + studentId;
+  newStudentStatsDivElement.classList.add('studentInHistory');
+
+  const studentGeneralStatsDivElement = document.createElement('div');
+  studentGeneralStatsDivElement.id = 'studentStats-' + studentId + '-generalStats';
+  studentGeneralStatsDivElement.innerText = studentId + ': ' + student.points + ' points. ' + student.numberAnswered + ' questions answered: ' + student.numberCorrect + ' correct, ' + student.numberWrong + ' wrong.';
+
+  newStudentStatsDivElement.appendChild(studentGeneralStatsDivElement);
+
+  studentsStatsDivElement.appendChild(newStudentStatsDivElement);
+}
+
+// Start listening to the student's individual answers
+function initStudentQuestionListener({ studentId, studentsCollectionRef } = {}) {
+  studentsCollectionRef.doc(studentId).collection('questions')
+  .onSnapshot(function(querySnapshot) {
+    querySnapshot.docChanges().forEach(function(change) {
+      // Only add the added answers, the past answers can't be updated
+      if (change.type === "added") {
+        // Add answers to student stats UI
+        addAnswerToStudentStats({ studentId, questionId:change.doc.id, answer: change.doc.data() });
+      }
+    });
+  });
+}
+
+// Add a student's individual answer to the student's history section
+function addAnswerToStudentStats({ studentId, questionId, answer } = {}) {
+  const singleStudentStatsDivElement = document.getElementById('studentStats-' + studentId);
+
+  const studentAnswerDivElement = document.createElement('div');
+  
+  const questionIdDivElement = document.createElement('div');
+  questionIdDivElement.innerText = 'Question with ID ' + questionId;
+
+  const questionTitleDivElement = document.createElement('div');
+  questionTitleDivElement.innerText = answer.title;
+
+  const questionAnswerDivElement = document.createElement('div');
+
+  if (answer.correct) {
+    questionAnswerDivElement.innerText = 'Correctly answered';
+  } else {
+    questionAnswerDivElement.innerText = 'Incorrect answer, chose: ' + answer.chosen;
```

Comment:
Maybe add some quotation marks to the question:
`questionAnswerDivElement.innerText = 'Incorrect answer, "' + answer.chosen; + '" chosen`

Reply:
Nice, done.